### PR TITLE
remove deprecated summary_info propery of ImageFileCollection

### PR DESCRIPTION
**Pull request by MSeifert04**
- [x] Did you add an entry to the "Changes.rst" file?
- [x] Does this PR add, rename, move or remove any existing functions or parameters?

The attribute is marked as deprecated in the docstring and it was since the `ImageFileCollection` was implemented in #217 . I had to change quite a lot to actually remove it.

### CODE REVIEW

File: ccdproc/image_collection.py [hunk 1]
```diff
@@ -142,18 +142,6 @@ def summary(self):
         return self._summary_info
 
     @property
-    def summary_info(self):
```
Comment (mwcraig) [inline ccdproc/image_collection.py]: Add astropy deprecation warning to function.

**Comment by MSeifert04** [comment]
The property and the test are restored. It will now warn the user if he accessed the property about the future removal.

**Comment by MSeifert04** [comment]
The tests passed. So this is ready for a final review 😄

**Comment by mwcraig** [comment]
Thanks again for catching this and cleaning it up...and sorry it made it into the codebase in the first place 😊

**Pull request merged**